src/Kernel.php

```diff
@@ -13,49 +13,41 @@
 
 namespace ApiPlatform\PDGBundle;
 
-use ApiPlatform\PDGBundle\DependencyInjection\ApiPlatformPDGExtension;
-use Symfony\Bundle\FrameworkBundle\FrameworkBundle;
-use Symfony\Bundle\FrameworkBundle\Kernel\MicroKernelTrait;
+use Symfony\Component\Config\FileLocator;
 use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\Console\Application;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
-use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+use Symfony\Component\DependencyInjection\Loader\XmlFileLoader;
+use Symfony\Component\DependencyInjection\Reference;
 use Symfony\Component\HttpKernel\Kernel as BaseKernel;
 
 class Kernel extends BaseKernel
 {
-    use MicroKernelTrait;
-
-    private function configureContainer(ContainerConfigurator $container, LoaderInterface $loader, ContainerBuilder $builder): void
-    {
-        $configDir = $this->getConfigDir();
-
-        $container->import($configDir.'/{packages}/*.{php,yaml}');
-        $container->import($configDir.'/{packages}/'.$this->environment.'/*.{php,yaml}');
-
-        $builder->registerExtension(new ApiPlatformPDGExtension());
-
-        $pdgConfig = getenv('PDG_CONFIG') ?? 'pdg.config.yaml';
-        if ($pdgConfig && is_file(getcwd().\DIRECTORY_SEPARATOR.$pdgConfig)) {
-            $container->import(getcwd().\DIRECTORY_SEPARATOR.$pdgConfig);
-        }
-
-        if (is_file($configDir.'/services.yaml')) {
-            $container->import($configDir.'/services.yaml');
-            $container->import($configDir.'/{services}_'.$this->environment.'.yaml');
-        } else {
-            $container->import($configDir.'/{services}.php');
-        }
-    }
-
-    public function getProjectDir(): string
+    public function registerBundles(): iterable
     {
-        return \dirname(__DIR__);
+        return [];
     }
 
-    public function registerBundles(): iterable
+    public function registerContainerConfiguration(LoaderInterface $loader)
     {
-        return [
-            new FrameworkBundle(),
-        ];
+        $loader->load(function (ContainerBuilder $container) {
+            // Load services
+            (new XmlFileLoader($container, new FileLocator(__DIR__.'/../config')))->load('services.xml');
```

Comment:
what about `        $configDir = $this->getConfigDir();` instead of `__DIR__` ?

Reply:
`Kernel::getConfigDir` does not exist by default